pkg/operator/ceph/object/bucket/predicate.go

```diff
@@ -62,8 +64,16 @@ func predicateController(ctx context.Context, c client.Client) predicate.Funcs {
 						}
 					}
 				}
-			}
 
+			case *cephv1.CephObjectStore:
+				objNew := e.ObjectNew.(*cephv1.CephObjectStore)
+				if objOld.Spec.Gateway.ExternalRgwEndpoints[0] != objNew.Spec.Gateway.ExternalRgwEndpoints[0] {
+					logger.Infof("there is an update event on CephObjectStore CR, so reconcile the bucket provisioner")
+
+					return true
```

Comment:
Have you tested that this works?

This returns `true` to indicate that the reconcile loop needs to be run, but that will mean the object sent to the reconcile loop is the CephObjectStore and not the OBCs associated with said store. At least, I think that's how it works.

I think this should either use a [MapFunc](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/handler#MapFunc) to enqueue the affected OBCs, or it should restart the OBC controller.

Reply:
@BlaineEXE looking for thoughts on this?

PLus the current implementation if you can review, I updated the watcher with handler if it's external reconcile, but not sure in the handler if we can compare old objects with newer one.

We can do ith predicate but the problem was what we were returning, can we return `[]reconcile.Request{}` from predicate,

Or some other instruction that will just restarts the object-bucket controller

Reply:
For testing, I tried using the same checks at csi controller and it worked.